### PR TITLE
Ion Rifles from R&D

### PR DESCRIPTION
**Pull request by brndd**
Ion rifles can't be acquired from anywhere currently. This adds them to R&D, which is where you also get laser rifles (so I think it makes more sense than cargo). They require the same materials as laser rifles (iron, glass and diamonds) and similar research, with some extra electromagnetic research—but it's not like research requirements really matter considering how easy they are to max out. They come in a locked box, like all other guns.

:cl:
 * rscadd: Ion rifles can now be created in R&D with similar material and research requirements as laser rifles, but with added electromagnetics research. As with other guns, they come in a locked box.

### CODE REVIEW

**Comment by boy2mantwicethefam** [comment]
I'm a science metaclubber and I approve of this PR

**Comment by boy2mantwicethefam** [comment]
Required electromagnetic research is very low IMO, what you get from throwing in an atmos analyzer, up it to 5

**Comment by SonixApache** [comment]
this has been the month of the powercreep

why do we need more ion guns?

**Comment by Kharkov72** [comment]
Because they only exist in secure armories in low numbers. There SHOULD be a way to actually get more.
Maybe make it take uranium as well though, since EMPnades need them and all.

**Comment by brndd** [comment]
I think it's stupid having relatively common items that can not be acquired anywhere except their roundstart locations. I will continue fixing these whenever I come across them when I deem it reasonable, and ions are IMO reasonable considering laser rifles (a similar scifi weapon that comes in ~~threes~~ ones in the armory) are also in R&D.

**Comment by SonixApache** [comment]
They're really not relatively common. Ion guns are rare, and probably intentionally so.
Why do they need to be common? Is them being rare an issue?

**Comment by brndd** [comment]
As suggested they now require 5 in magnet tech and some (8000 units) of uranium to make. I reduced the diamond requirement to compensate for the added uranium.

**Comment by Kharkov72** [comment]
What's wrong with them being common? It's a damage type that only has certain select uses, is huge so you either need to use armor, your back, or an entire BoH slot to hold them, so it's not convenient to mass produce anyways, and more EMPs come from shards than actual weaponized sources.
This could encourage more use of ions.

**Comment by brndd** [comment]
>They're really not relatively common.

But they are relatively common. You can get them from the armoury every roundstart by taking ten steps as the warden. There's also only one laser rifle in the armoury, making it even rarer than ion guns, yet you can make more in R&D. And as @Kharkov72 said, ion rifles aren't terribly powerful, having a relatively niche use. Certainly they're not nearly as powerful as the laser rifle.

**Comment by iFlashYou** [comment]
why not the ion pistol?

**Comment by SonixApache** [comment]
Ion guns are extremely powerful if used correctly, if only because they disrupt comms.

They're not common if there's 3 (three) of them available *with no way to get more if you lose them*.
As for the laser *rifle*, that's a box-specific thing, as a "you have one high-tech weapon in stock" type of thing.

@iFlashYou the creep is absurd

**Comment by brndd** [comment]
>why not the ion pistol?

I was told the ion pistol is an adminbus item. To my knowledge it's not mapped in anywhere.

**Comment by Kitfox88** [comment]
mecha metaclub mad about this pr

**Comment by SonixApache** [comment]
The ion pistol is mapped in packed, for some reason.
If you need to so desperately add ions to R&D, why not make an ion carbine or something? 
As a sort of stop-gap between the pistol and full rifle.

**Comment by iFlashYou** [comment]
whats wrong with the pistol?

**Comment by brndd** [comment]
> If you need to so desperately add ions to R&D, why not make an ion carbine or something?
> As a sort of stop-gap between the pistol and full rifle.

Would an ion carbine not be even more powerful, since you could fit it in a backpack?

**Comment by SonixApache** [comment]
I mean, yes, but if you make it a slightly upcapped ion pistol, not really.
Adding straight copies of whatever's already in the armory at roundstart to the R&D section is kind of lame.
(R&D laser rifle excluded obviously)

**Comment by brndd** [comment]
I don't really see why. The ion pistol gets half the shots (5 vs. 10) of the ion rifle, but is a medium-size item so it fits in your backpack and secbelt. For the purposes that you think ions are strong (disabling comms), I'd argue the ion pistol is *stronger* than the rifle because it's easier to carry, and the fewer shots don't really matter because one EMP is enough to turn off someone's headset.

**Comment by SonixApache** [comment]
Yes, the ion pistol is a lot stronger than the rifle in that regard.
Ions are the ultimate sec killer, since they kill their headsets, SPSes, and stun weaponry.
They're also the ultimate borg killer, and supposedly they fuck up mechs too.

I'm saying that
 - Ion carbine
 - 7 shots a mag
 - low power EMP
 - medium size so it fits in pack / on belt (but not IN a belt)
 - not a handgun

Would be an okay thing to add to R&D. Maybe. Uniqueness, y'know.

**Comment by iFlashYou** [comment]
>dude it kills poor red man

:^)

**Comment by brndd** [comment]
But this is on a weapon that's very annoying to carry around. It only goes in your suit storage or hands. I'm not fundamentally opposed to an ion carbine, but I'm not going to go through the trouble of implementing it (90% of said trouble being spritework) unless it's really deemed to be the better option.

File: code/modules/research/designs/weapons.dm [hunk 1]
```diff
@@ -94,6 +94,18 @@
 	locked = TRUE
 	req_lock_access = list(access_armory, access_weapons)
 
+/datum/design/ionrifle
+	name = "Ion Rifle"
+	desc = "A man portable anti-armor weapon designed to disable mechanical threats."
```
Comment (MadmanMartian) [inline code/modules/research/designs/weapons.dm:99]: This has no armor penetration, why have its description be anti-armor?

Or is this the description of the Ion Rifle?

Comment (brndd) [reply code/modules/research/designs/weapons.dm:99]: It's copypasted from the ion rifle.

Comment (Kitfox88) [reply code/modules/research/designs/weapons.dm:99]: anti-armor as in anti-vehicle, not armor piercing, i presume

Comment (SonixApache) [reply code/modules/research/designs/weapons.dm:99]: armor as in mechas

Comment (MadmanMartian) [reply code/modules/research/designs/weapons.dm:99]: Huh

**Comment by brndd** [comment]
@SonixApache is adding ion carbines, which will replace this PR and be an item unique to R&D. They'll be something of a middle ground between the ion pistol and the ion rifle, having fewer shots than the rifle and not fitting on belts like the pistol. I'm closing this PR since he's putting his version up after he's done with the inhand sprites.

**Comment by Hoshi-chan** [comment]
Ion rifles are absolutely limited in number by design. I forget when and how this came about, but the reason ion rifles are armory-only, can only be carried on the back slot, and are unable to be duplicated, is a deliberate balancing decision because of how strong and disruptive they are.

It's also why ion pistols are bus-only.

**Pull request closed without merging**